### DISCUSSION TITLE
ChatGPT应用开发小记 · BMPI

### DISCUSSION THREAD

**Discussion by giscus** [bmpi-dev/bmpi.dev#144] [Announcements]
# ChatGPT应用开发小记 · BMPI

本文分享了笔者使用 chatGPT 开发 myGPTReader 开源项目的过程。

https://www.bmpi.dev/dev/chatgpt-development-notes/my-gpt-reader/

**Comment by madawei2699** [comment]
进一步阅读材料👇

**Reply by madawei2699** [reply]
https://www.pinecone.io/learn/chunking-strategies/

**Reply by madawei2699** [reply]
ChatGPT的前世今生：http://lipiji.com/slides/ChatGPT_ppf.pdf

**Reply by madawei2699** [reply]
LLM不是搜索引擎，而是推理引擎：https://eugeneyan.com/writing/llm-experiments/

**Reply by madawei2699** [reply]
提示语工程好的入门材料：https://www.promptingguide.ai/zh

**Reply by madawei2699** [reply]
可视化展示 langchain 的工作流：https://github.com/logspace-ai/langflow

**Reply by madawei2699** [reply]
AI 应用工程化的问题：https://huyenchip.com/2023/04/11/llm-engineering.html

**Reply by madawei2699** [reply]
[Autonomous Agents & Agent Simulations](https://blog.langchain.dev/agents-round/)

**Reply by madawei2699** [reply]
Embedding很好的介绍：https://github.com/lvwzhen/law-cn-ai#-runtime

**Reply by madawei2699** [reply]
https://note.com/mahlab/n/n7d72e83904cc

**Reply by madawei2699** [reply]
[Improving Document Retrieval with Contextual Compression](https://blog.langchain.dev/improving-document-retrieval-with-contextual-compression/)

**Reply by madawei2699** [reply]
## 图书推荐
动手学深度学习：https://zh.d2l.ai/

**Reply by madawei2699** [reply]
# 课程推荐
https://www.deeplearning.ai/short-courses/chatgpt-prompt-engineering-for-developers/

**Comment by decli** [comment]
马老师，请教下文章中的时序图是用什么工具或者框架根据 markdown 转换出来的吗？

**Reply by madawei2699** [reply]
https://www.bmpi.dev/self/my-drawing-toolbox/ 看这篇文章介绍
